app/src/main/java/com/example/anfcodetest/CustomViewHolder.kt

```diff
@@ -0,0 +1,61 @@
+package com.example.anfcodetest
+
+import android.content.Intent
+import android.text.Html
+import android.text.method.LinkMovementMethod
+import android.view.View
+import android.widget.Button
+import androidx.core.content.ContextCompat.startActivity
+import androidx.recyclerview.widget.RecyclerView
+import com.squareup.picasso.Picasso
+import kotlinx.android.synthetic.main.card_row.view.*
+
+class CustomViewHolder(private val view: View) : RecyclerView.ViewHolder(view) {
+
+    fun bind(card: Card) = with(view) {
+        loadBackgroundImage(card.backgroundImage)
+        title.setTextAndVisibility(card.title)
+        promoMessage.setTextAndVisibility(card.promoMessage)
+        topDescription.setTextAndVisibility(card.topDescription)
+        setUpBottomDescription(card.bottomDescription)
+        topButton.setUpButton(card.content?.firstOrNull())
+        if (card.content?.size == 2) {
+            bottomButton.setUpButton(card.content.get(1))
+        }
+    }
+
+    private fun onButtonClicked(title: String?, target: String?) = with(view) {
+        val intent = Intent(bottomButton.context, ButtonTargetActivity::class.java)
+            .putExtra("title", title)
```

Comment:
We can get rid of recalling the intent each time we want to put a new extra and just chain them on to our `Intent` constructor here. So instead of 

`val intent = Intent(bottomButton.context, ButtonTargetActivity::class.java)
intent.putExtra("title", title)`

We can just have this!